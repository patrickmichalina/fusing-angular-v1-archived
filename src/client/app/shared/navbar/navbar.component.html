<nav>
  <ul *ngIf="userView$ | async as user" >
    <li><a routerLink title="I am lazy also!">Home</a></li>
    <li><a routerLink="about" title="I am a lazy module for the about page">Performance</a></li>
    <li><a routerLink="demos" title="I am a lazy module for the demos page">Isomorphic</a></li>
    <li><a href="/api-docs" title="View the rest API docs" pmExternalLink>API Docs</a></li>
    <li><a href="https://github.com/patrickmichalina/fusing-angular" pmExternalLink>View on Github</a></li>
    <li [ngSwitch]="user.show">
      <div *ngSwitchCase="0">
        <div>{{ user.greeting }} </div> 
        <div>
          <span>You have the following roles</span>
          <ul>
            <li *ngFor="let role of user.roles; trackBy: trackByRole">{{ role }}</li>
          </ul>
        </div> 
        <button (click)="auth.logout()">Logout</button>
      </div>
      <div *ngSwitchCase="1">
        <button (click)="login()">Login</button>
      </div>
    </li>
  </ul>
</nav>