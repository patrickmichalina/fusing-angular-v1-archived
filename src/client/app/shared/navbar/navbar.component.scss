@import '../../../styles/variables';
:host {
  display: block;
  z-index: 10;
  position: fixed;
  left: 0;
  right: 0;
  .fx-1 {
    flex: 1;
  }
  @media screen and (max-width: 700px) {
    div {
      display: none;
    }
  }
  mat-toolbar {
    height: $navbar-height;
  }
}
