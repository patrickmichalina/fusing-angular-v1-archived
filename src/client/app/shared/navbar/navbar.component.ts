import { ChangeDetectionStrategy, Component } from '@angular/core'

export interface User {
  readonly photoURL: string
  readonly email: string
  readonly name: string
}

@Component({
  selector: 'pm-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class NavbarComponent {
}
