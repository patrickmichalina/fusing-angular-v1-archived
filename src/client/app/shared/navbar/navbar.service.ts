import { Observable } from 'rxjs/Observable'
import { Injectable } from '@angular/core'
import { of } from 'rxjs/observable/of'

export interface INavbarService {
  readonly menu$: Observable<ReadonlyArray<any>>
}

@Injectable()
export class NavbarService implements INavbarService {
  readonly menu$ = of([
    { route: 'dashboard', name: 'Dashboard' },
    { route: 'about', name: 'About' }
  ])
}
