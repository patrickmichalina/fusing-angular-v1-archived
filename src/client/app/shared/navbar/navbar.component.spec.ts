import { RouterTestingModule } from '@angular/router/testing'
import { NavbarComponent } from './navbar.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { NavbarService } from './navbar.service'
import { Component } from '@angular/core'

describe(NavbarComponent.name, () => {
  let fixture: ComponentFixture<NavbarComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [NavbarComponent, TestComponent],
      providers: [NavbarService]
    }).compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(NavbarComponent)
  })

  afterEach(() => {
    TestBed.resetTestingModule()
  })

  it('should compile', async(() => {
    fixture.detectChanges()
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture.nativeElement).toMatchSnapshot()
  }))
})

@Component({
  selector: 'test-component',
  template: '<pm-navbar></pm-navbar>'
})
class TestComponent {}
