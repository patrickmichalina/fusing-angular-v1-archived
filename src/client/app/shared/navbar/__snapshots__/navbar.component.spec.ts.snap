// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent should compile 1`] = `
<div
  id="root0"
  ng-version="6.0.0-beta.7"
>
  <nav>
    
    <ul>
      <li>
        <a
          href="/"
          ng-reflect-router-link=""
          routerlink=""
          title="I am lazy also!"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
          ng-reflect-router-link="about"
          routerlink="about"
          title="I am a lazy module for the about page"
        >
          Performance
        </a>
      </li>
      <li>
        <a
          href="/demos"
          ng-reflect-router-link="demos"
          routerlink="demos"
          title="I am a lazy module for the demos page"
        >
          Isomorphic
        </a>
      </li>
      <li>
        <a
          href="/api-docs"
          pmexternallink=""
          title="View the rest API docs"
        >
          API Docs
        </a>
      </li>
      <li>
        <a
          href="https://github.com/patrickmichalina/fusing-angular"
          pmexternallink=""
        >
          View on Github
        </a>
      </li>
      <li
        ng-reflect-ng-switch="0"
      >
        
        <div>
          <div>
            Welcome, undefined 
          </div>
          <div>
            <span>
              You have the following roles
            </span>
            <ul>
              
            </ul>
          </div>
          <button>
            Logout
          </button>
        </div>
        
      </li>
    </ul>
  </nav>
</div>
`;
